util/noeintr_close.c

```diff
@@ -0,0 +1,284 @@
+/**
+ * Goal: portably close a file descriptor in a multithreaded process without
+ * running into problems with EINTR.
```

Comment:
I would add "POSIX specifies that close(2) may be interrupted by a signal and return -1 with errno set to EINTR, but explicitly does not specify whether the file descriptor has been closed if that occurs".